cockatrice/src/carddatabase.cpp

```diff
@@ -341,39 +341,35 @@ static QXmlStreamWriter &operator<<(QXmlStreamWriter &xml, const CardInfo *info)
 }
 
 CardDatabase::CardDatabase(QObject *parent)
-    : QObject(parent), noCard(0), loadStatus(NotLoaded)
+    : QObject(parent), loadStatus(NotLoaded)
 {
-    connect(settingsCache, SIGNAL(cardDatabasePathChanged()), this, SLOT(loadCardDatabase()));
-    connect(settingsCache, SIGNAL(tokenDatabasePathChanged()), this, SLOT(loadTokenDatabase()));
-
-    noCard = new CardInfo();
+    connect(settingsCache, SIGNAL(cardDatabasePathChanged()), this, SLOT(loadCardDatabases()));
 }
 
 CardDatabase::~CardDatabase()
 {
     clear();
-    delete noCard;
 }
 
 void CardDatabase::clear()
 {
-    QHashIterator<QString, CardSet *> setIt(sets);
-    while (setIt.hasNext()) {
-        setIt.next();
-        delete setIt.value();
-    }
-    sets.clear();
-    
     QHashIterator<QString, CardInfo *> i(cards);
     while (i.hasNext()) {
         i.next();
-        delete i.value();
+        removeCard(i.value());
+        i.value()->deleteLater();
     }
-    cards.clear();
 
-    // The pointers themselves were already deleted, so we don't delete them
-    // again.
+    // The pointers themselves were already deleted, so we don't delete them again.
+    cards.clear();
     simpleNameCards.clear();
+
+    QHashIterator<QString, CardSet *> setIt(sets);
+    while (setIt.hasNext()) {
+        setIt.next();
+        delete setIt.value();
```

Comment:
CardSet is a QList; a QList doesn't inherit from QObject, so there's no deleteLater() method available.